src/ray/raylet/scheduling/cluster_resource_scheduler.h

```diff
@@ -64,36 +65,22 @@ class ClusterResourceScheduler {
   ///  Find a node in the cluster on which we can schedule a given resource request.
   ///  In hybrid mode, see `scheduling_policy.h` for a description of the policy.
   ///
-  ///  \param resource_request: Task to be scheduled.
-  ///  \param scheduling_strategy: Strategy about how to schedule this task.
-  ///  \param actor_creation: True if this is an actor creation task.
-  ///  \param force_spillback: For non-actor creation requests, pick a remote
-  ///  feasible node. If this is false, then the task may be scheduled to the
-  ///  local node.
-  ///  \param violations: The number of soft constraint violations associated
-  ///                     with the node returned by this function (assuming
-  ///                     a node that can schedule resource_request is found).
-  ///  \param is_infeasible[in]: It is set true if the task is not schedulable because it
-  ///  is infeasible.
+  ///  \param task_spec: Task/Actor to be scheduled.
+  ///  \param prioritize_local_node: true if we want to try out local node first.
+  ///  \param exclude_local_node: true if we want to avoid local node. This will cancel
+  ///  prioritize_local_node if set to true.
+  ///  \param requires_object_store_memory: take
+  ///  object store memory usage as part of scheduling decision. \param
+  ///  is_infeasible[out]: It is set true if the task is not schedulable because it is
+  ///  infeasible.
   ///
-  ///  \return -1, if no node can schedule the current request; otherwise,
-  ///          return the ID of a node that can schedule the resource request.
-  int64_t GetBestSchedulableNode(const ResourceRequest &resource_request,
-                                 const rpc::SchedulingStrategy &scheduling_strategy,
-                                 bool actor_creation, bool force_spillback,
-                                 int64_t *violations, bool *is_infeasible);
+  ///  \return emptry string, if no node can schedule the current request; otherwise,
+  ///          return the string name of a node that can schedule the resource request.
 
-  /// Similar to
-  ///    int64_t GetBestSchedulableNode(...)
-  /// but the return value is different:
-  /// \return "", if no node can schedule the current request; otherwise,
-  ///          return the ID in string format of a node that can schedule the
-  //           resource request.
-  std::string GetBestSchedulableNode(
-      const absl::flat_hash_map<std::string, double> &resource_request,
-      const rpc::SchedulingStrategy &scheduling_strategy,
-      bool requires_object_store_memory, bool actor_creation, bool force_spillback,
-      int64_t *violations, bool *is_infeasible);
+  std::string GetBestSchedulableNode(const TaskSpecification &task_spec,
+                                     bool prioritize_local_node, bool exclude_local_node,
```

Comment:
These two flags seem hard to understand. Shouldn't we resolve this on the caller's side?

Reply:
Could you be more specific?

Reply:
In the comment it says, if exclude_local_node=true, it'll ignore prioritize_local_node.

I feel it's wired that we have two flags here to control whether we'd like to let the task schedule be on the local node or not. Definitely one of them is invalid: prioritize_local_node=True and exclude_local_node=True.

Also, it might be better if we can have a Policy class (maybe in the future) for all the setup.

Reply:
yeah, I think we need to refine it a bit, especially after we merge with gcs resource scheduler.